### PR TITLE
Menu

### PR DESCRIPTION
**Pull request by fabientownsend**

### CODE REVIEW

File: lib/menu.ex [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  end
+
+  defp games do
+    [
+      {"0 - human vs human", [Human, Human]},
+      {"1 - human vs computer", [Human, Computer]},
+      {"2 - computer vs human", [Computer, Human]},
+      {"3 - computer vs computer", [Computer, Computer]}
+    ]
+  end
```
Comment (fabientownsend) [inline lib/menu.ex]: Not really sure if it belongs here or in the factory

File: lib/cli.ex [hunk 2]
```diff
@@ -0,0 +1,21 @@
+defmodule Cli do
+  def get_integer, do: get_integer("")
+  def get_integer(request) do
+    String.trim(IO.gets request)
```
Comment (felipesere) [inline lib/cli.ex]: Why oh why isn't this:
request
|> IO.gets
|> String.trim
|> ...

Mixing and matching looks off to me... asymmetrical!

File: lib/cli.ex [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    case Integer.parse(input) do
+      {position, ""} -> position
+      _ -> :bad_parse
+    end
+  end
+
+  defp validate_integer(:bad_parse) do
+    get_integer("The value must be an integer")
+  end
+
```
Comment (felipesere) [inline lib/cli.ex]: No empty lines for functions that are supposed to be the same 😄

File: lib/menu.ex [hunk 4]
```diff
@@ -0,0 +1,36 @@
+defmodule Menu do
+  def game_types do
+    display_game_types
+
+    get_value_betweenf(0, Enum.count(games) - 1)
```
Comment (felipesere) [inline lib/menu.ex]: `betweenf` ?
And I would probably use a variable to explain `Enum.count(games) - 1`

Comment (felipesere) [reply lib/menu.ex]: And please use `games()` otherwise its hard to tell that it is a function, not a variable

File: lib/menu.ex [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    |> get_players
+  end
+
+  defp display_game_types do
+    IO.puts Enum.map(games(), fn({game_type, _}) -> game_type <> "\n" end)
```
Comment (felipesere) [inline lib/menu.ex]: This could be:
games()
|> Enum.map(fn({type, _}) -> type end)
|> Enum.join("\n")
|> IO.put

Comment (felipesere) [reply lib/menu.ex]: and you could probably extract the anonymous function into a real function and just write
games
|> Enum.map(to_type/1)
|> Enum.join("\n")
|> IO.puts

File: lib/menu.ex (hunk 4 above)
Comment (fabientownsend) [reply lib/menu.ex]: My bad, wasn't the last commit!

File: lib/menu.ex (hunk 5 above)
Comment (fabientownsend) [reply lib/menu.ex]: Clearly much better!

**Comment by felipesere** [comment]
Looking spiffy. Do you feel like you are getting a handle of good Elixir style?

**Pull request merged**